timesketch/lib/analyzers/dfiq_plugins/manager.py

```diff
@@ -0,0 +1,308 @@
+# Copyright 2024 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This file contains a class for managing DFIQ analyzers."""
+
+import importlib
+import inspect
+import json
+import logging
+import os
+
+from timesketch.lib.aggregators import manager as aggregator_manager
+from timesketch.lib.analyzers import interface
+from timesketch.lib.analyzers import manager as analyzer_manager
+from timesketch.models.sketch import Timeline
+
+
+logger = logging.getLogger("timesketch.analyzers.dfiq_plugins.manager")
+
+
+def load_dfiq_analyzers():
+    """Loads DFIQ analyzer classes."""
+
+    DFIQ_ANALYZER_PATH = os.path.dirname(os.path.abspath(__file__))
+
+    # Clear existing registrations before reloading
+    for name, analyzer_class in analyzer_manager.AnalysisManager.get_analyzers(
+        include_dfiq=True
+    ):
+        if (
+            hasattr(analyzer_class, "IS_DFIQ_ANALYZER")
+            and analyzer_class.IS_DFIQ_ANALYZER
+        ):
+            try:
+                analyzer_manager.AnalysisManager.deregister_analyzer(name)
+                logger.info("Deregistered DFIQ analyzer: %s", name)
+            except KeyError as e:
+                logger.error(str(e))
+
+    # Dynamically load DFIQ Analyzers
+    for filename in os.listdir(DFIQ_ANALYZER_PATH):
+        if filename.endswith(".py") and not filename.startswith("__"):
+            module_name = filename[:-3]  # Remove .py extension
```

Comment:
Consider using path.splitext instead. It handles filenames with multiple dots etc and is a bit more robust.
```suggestion
            module_name = os.path.splitext(filename)[0]  # Remove .py extension
```